### PR TITLE
Convert Checkbox to Typescript

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

Converts `Checkbox` to Typescript

### Related issue(s)/PR(s)

https://github.com/strapi/design-system/issues/983

### CODE REVIEW

**Comment by joshuaellis** [comment]
You've not changed the e2e test to TS by the way.

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx
Comment (joshuaellis) [inline packages/strapi-design-system/src/Checkbox/Checkbox.tsx:1]: You've not typed `CheckboxTick`

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -24,10 +23,18 @@ const CheckboxTick = (props) => {
   return <BaseCheckbox id={id} {...props} />;
 };
 
-export const Checkbox = ({ children, disabled, id, hint, error, ...props }) => {
+interface CheckboxProps {
+  children: React.ReactNode;
+  disabled?: boolean;
+  error?: string;
+  hint?: string | React.ReactNode | React.ReactNode[];
+  id?: string | number;
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/Checkbox/Checkbox.tsx:31]: these can all be brought from `FieldProps` if you `extend` this type

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -24,10 +23,18 @@ const CheckboxTick = (props) => {
   return <BaseCheckbox id={id} {...props} />;
 };
 
-export const Checkbox = ({ children, disabled, id, hint, error, ...props }) => {
+interface CheckboxProps {
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/Checkbox/Checkbox.tsx:26]: This should `extend BaseCheckboxProps` because we spread the rest of props to it.

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx:31]: ok thanks

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx
Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx:1]: yes thanks

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx (hunk 2 above)
Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx]: ok done thanks

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx
Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx:1]: done

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -1,33 +1,39 @@
...
 
   return <BaseCheckbox id={id} {...props} />;
 };
 
-export const Checkbox = ({ children, disabled, id, hint, error, ...props }) => {
+interface CheckboxProps extends BaseCheckboxProps {
+  children: React.ReactNode;
+  disabled?: boolean;
+  hint?: string | React.ReactNode | React.ReactNode[];
+  error?: string;
```
Comment (joshuaellis) [inline packages/strapi-design-system/src/Checkbox/Checkbox.tsx:30]: i think these should just be extended from `FieldProps` imo

Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx:30]: ok thanks

File: packages/strapi-design-system/src/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -1,33 +1,39 @@
...
-import { BaseCheckbox } from '../BaseCheckbox';
+import { BaseCheckbox, BaseCheckboxProps } from '../BaseCheckbox';
 import { Box } from '../Box';
 import { Field, FieldHint, FieldError, useField } from '../Field';
 import { Flex } from '../Flex';
 import { useId } from '../hooks/useId';
 import { Typography } from '../Typography';
 
-const CheckboxLabel = styled(Typography)`
+const CheckboxLabel = styled(Typography)<{ disabled: boolean }>`
```
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/strapi-design-system/src/Checkbox/Checkbox.tsx:12]: done thanks

**Review by joshuaellis — APPROVED** [review]
Great work ⭐

**Review by christiancp100 — APPROVED** [review]

**Pull request merged**